experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -666,182 +460,52 @@ export class ComposeQueue<T> {
 		} else if (newMark === undefined) {
 			const length = getOutputLength(baseMark);
 			return this.dequeueBase(length);
-		} else if (isAttach(newMark)) {
-			const newRev = newMark.revision ?? this.newRevision;
-			if (
-				isDetachMark(baseMark) &&
-				areInverseRevisions(
-					newRev,
-					baseMark.revision ?? this.baseMarks.revision,
-					this.revisionMetadata,
-				)
-			) {
-				const baseMarkLength = getInputLength(baseMark);
-				const newMarkLength = getOutputLength(newMark);
-				// There is some change foo that is being cancelled out as part of a rebase sandwich.
-				// The marks that make up this change (and its inverse) may be broken up differently between the base
-				// changeset and the new changeset because either changeset may have been composed with other changes
-				// whose marks may now be interleaved with the marks that represent foo/its inverse.
-				// This means that the base and new marks may not be of the same length.
-				// We do however know that the all of the marks for foo will appear in the base changeset and all of the
-				// marks for the inverse of foo will appear in the new changeset, so we can be confident that whenever
-				// we encounter such pairs of marks, they do line up such that they describe changes to the same first
-				// cell. This means we can safely treat them as inverses of one another.
-				if (newMarkLength < baseMarkLength) {
-					baseMark = this.baseMarks.dequeueInput(newMarkLength);
-					newMark = this.newMarks.dequeue();
-				} else if (newMarkLength > baseMarkLength) {
-					baseMark = this.baseMarks.dequeue();
-					newMark = this.newMarks.dequeueOutput(baseMarkLength, true);
-				} else {
-					baseMark = this.baseMarks.dequeue();
-					newMark = this.newMarks.dequeue();
+		} else if (areOutputCellsEmpty(baseMark) && areInputCellsEmpty(newMark)) {
+			assert(isExistingCellMark(baseMark), "Only existing cell mark can have empty output");
+			let baseCellId: DetachEvent;
+			if (markEmptiesCells(baseMark)) {
+				assert(isDetachMark(baseMark), "Only detach marks can empty cells");
+				const baseRevision = baseMark.revision ?? this.baseMarks.revision;
+				const baseIntention = getIntention(baseRevision, this.revisionMetadata);
+				if (baseRevision === undefined || baseIntention === undefined) {
+					// This case should only happen when squashing a transaction.
+					assert(isNewAttach(newMark), "Unhandled case");
```

Comment:
In the future, we want to support reattaches in the new change here.
We will need to be able to order the base mark relative to the new mark by looking at the lineage of the new mark (which will be obtained by rebasing the reattach over interim changes (which requires the local changes to have a revision tag))